dnstable/entry.c

```diff
@@ -378,6 +394,29 @@ dnstable_entry_to_json_fmt(const struct dnstable_entry *e,
 
 		status = yajl_gen_array_close(g);
 		assert(status == yajl_gen_status_ok);
+
+		if (add_raw_rdata) {
+			add_yajl_string(g, "rdata_raw");
+
+			status = yajl_gen_array_open(g);
+			assert(status == yajl_gen_status_ok);
+
+			for (size_t i = 0; i < n_rdatas; i++) {
+				wdns_rdata_t *rdata = rdata_vec_value(e->rdatas, i);
+				ubuf *rbuf = ubuf_init(2 * rdata->len + 1);
+				uint8_t *rbuf_as_str = NULL;
+				size_t rbuf_as_str_len = 0;
+
+				fmt_hex_str(rbuf, (char*)rdata->data, rdata->len);
+				ubuf_cterm(rbuf);
+				ubuf_detach(rbuf, &rbuf_as_str, &rbuf_as_str_len);
+				add_yajl_string(g, (char*)rbuf_as_str);
```

Comment:
We should consider appending the varint-formatted rrtype here, for efficiency of an ensuing raw query. The code for this would be roughly:

  uint8_t rrtype[3];   /* 16 bit rrtype will be at most 3 bytes of varint */
  fmt_hex_str(rbuf, rrtype, mtbl_varint_encode32(rrtype, e->rrtype));

Reply:
`dnsdbq -N 036e7331076578616d706c6503636f6d00 -j -l 0` gets 228296 rows.
appending the rrtype
`dnsdbq -N 036e7331076578616d706c6503636f6d0002 -j -l 0` gets 182804 rows.